### DISCUSSION TITLE
How to add a new Subsection to an existing part of the Configuration

### DISCUSSION THREAD

**Discussion by marianne013** [DIRACGrid/DIRAC#6295] [Q&A]
While the code below kind of works, I get the feeling this is not how it's intended.
The subsection we had in mind would exits as a handy dictionary :-S
```
from diraccfg.cfg import CFG
from DIRAC.ConfigurationSystem.Client.CSAPI import CSAPI
from DIRAC.ConfigurationSystem.Client.Config import gConfig

[...]

       # add a dummy value                                                                                                                                                                                            
        storageElements["UKI-LT2-IC-HEP-disk"]["dummy"] = "test"
        cfg = CFG()
        cfg.loadFromDict(storageElements)
        csapi.mergeCFGUnderSection("/Resources/StorageElements", cfg)
        result = csapi.commit()
        if not result["OK"]:
            log.error("OMG")
            return result
        return S_OK()
```

**Comment by fstagni** [comment]
What is the result you are getting? (and what were you hoping to get)

**Comment by marianne013** [comment]
Unfortunately I haven't had much time to look at DIRAC recently. I *think* (maybe @sfayer has a better memory) what we were trying to do was the opposite of: 
storageElementsResult = gConfig.getOptionsDictRecursively("/Resources/StorageElements")
Now I have a dictionary.
Then I update this dictionary with new elements.
How do I commit it back ?

**Reply by sfayer** [reply]
Yes, I think the basic question is "is there a helper function to merge a dictionary into a specific point in the config?" (and if the answer is no, I guess the follow up question will be "can we add one?" :-)

**Reply by fstagni** [reply]
From the CS WebApp you can "create a subsection" which accepts a "Config", is it what effectively you are trying to do here?
